java/core/src/java/org/apache/orc/impl/reader/StripePlanner.java

```diff
@@ -445,55 +509,83 @@ private static boolean hasSomeRowGroups(boolean[] includedRowGroups) {
    * @return the list of disk  ranges that will be loaded
    */
   private BufferChunkList planPartialDataReading(OrcIndex index,
-                                                 @NotNull boolean[] includedRowGroups) {
+                                                 @NotNull boolean[] includedRowGroups,
+                                                 int startGroup,
+                                                 TypeReader.ReadPhase readPhase) {
     BufferChunkList result = new BufferChunkList();
     if (hasSomeRowGroups(includedRowGroups)) {
       InStream.StreamOptions compression = dataReader.getCompressionOptions();
       boolean isCompressed = compression.getCodec() != null;
       int bufferSize = compression.getBufferSize();
       OrcProto.RowIndex[] rowIndex = index.getRowGroupIndex();
       for (StreamInformation stream : dataStreams) {
-        if (RecordReaderUtils.isDictionary(stream.kind, encodings[stream.column])) {
-          addChunk(result, stream, stream.offset, stream.length);
+        if (readPhase == TypeReader.ReadPhase.ALL
+            ||(readPhase == TypeReader.ReadPhase.LEADERS
```

Comment:
nit. space after || 
nit. reduce new lines

Reply:
> nit. space after ||
done

> nit. lets keep conditions on a single line
brought the subcondition into a single line not the entire condition as it extends the line boundary significantly

> nit. reduce new lines (e.g., processStream call)
Fixed the processStream call however I am unclear on how to handle parameters. I have generally been following chompdown if long, I see both variants in the code chomp or just wrap.

Reply:
Looks good now, thanks!